demos/common/python/models/bert.py

```diff
@@ -0,0 +1,196 @@
+        self.token_cls = [vocab['[CLS]']]
+        self.token_sep = [vocab['[SEP]']]
+        self.token_pad = [vocab['[PAD]']]
+
+    def check_input_output_names(self, input_names, output_names=None):
+        self.input_names = [i.strip() for i in input_names.split(',')]
+        if self.net.input_info.keys() != set(self.input_names):
+            raise RuntimeError('The demo expects input names: {}, actual network input names: {}'.format(
+                self.input_names, list(self.net.input_info.keys())))
+
+        self.max_length = self.net.input_info[self.input_names[0]].input_data.shape[1]
+
+        if output_names is None:
+            return
+
+        self.output_names = [o.strip() for o in output_names.split(',')]
+        if self.net.outputs.keys() != set(self.output_names):
+            raise RuntimeError('The demo expects output names: {}, actual network output names: {}'.format(
+                self.output_names, list(self.net.outputs.keys())))
+
+    def preprocess(self, inputs):
+        input_ids, attention_mask, token_type_ids = self.form_request(inputs)
+
+        pad_len = self.pad_input(input_ids, attention_mask, token_type_ids)
+        meta = {'pad_len': pad_len, 'inputs': inputs}
+
+        return self.create_input_array(input_ids, attention_mask, token_type_ids), meta
+
+    def pad_input(self, input_ids, attention_mask, token_type_ids):
+        pad_len = self.max_length - len(input_ids)
+        if pad_len < 0:
+            raise ValueError("The input request is longer than max number of tokens ({})"
+                             " processed by model".format(self.max_length))
+        input_ids += self.token_pad * pad_len
+        token_type_ids += self.token_pad * pad_len
+        attention_mask += self.token_pad * pad_len
+        return pad_len
+
+    def create_input_array(self, input_ids, attention_mask, token_type_ids):
+        inputs = {
+            self.input_names[0]: np.array([input_ids], dtype=np.int32),
+            self.input_names[1]: np.array([attention_mask], dtype=np.int32),
+            self.input_names[2]: np.array([token_type_ids], dtype=np.int32),
+        }
+        if len(self.input_names) > 3:
+            inputs[self.input_names[3]] = np.arange(len(input_ids), dtype=np.int32)[None, :]
+
+        return inputs
+
+    def reshape(self, new_length):
+        new_shapes = {}
+        for input_name, input_info in self.net.input_info.items():
+            new_shapes[input_name] = [1, new_length]
+        try:
+            default_input_shape = input_info.input_data.shape
+            self.net.reshape(new_shapes)
+            self.logger.debug("\tReshape model from {} to {}".format(default_input_shape, new_shapes[input_name]))
+            self.max_length = new_length
+        except RuntimeError:
+            raise RuntimeError("Failed to reshape the model")
+
+
+class NamedEntityRecognition(Bert):
+
+    def __init__(self, ie, model_path, vocab, input_names):
```

Comment:
Looks like the `input_names` are the `Bert` class parameter

Reply:
Updated. Also, `check_input_output_names` function was split to `check_input_names()` and `check_output_names()`.